src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -824,5 +824,35 @@ internal static StringComparison StringComparison
                     StringComparison.OrdinalIgnoreCase;
             }
         }
+
+        /// <summary>
+        /// Trims the ending directory separator if present.
+        /// </summary>
+        /// <param name="path"></param>
+        public static string TrimEndingDirectorySeparator(string path) =>
+            EndsInDirectorySeparator(path) && !PathInternal.IsRoot(path.AsSpan()) ?
+                path.Substring(0, path.Length - 1) :
+                path;
+
+        /// <summary>
+        /// Trims the ending directory separator if present.
```

Comment:
nit (for above too):

`Trims any trailing separators beyond the root of the path.`

Reply:
done